File: SourceCppProject/sources/Models/LinesStorageMap.cpp
#include "LinesStorageMap.hpp"


LinesStorageMap::LinesStorageMap()
{
}

LinesStorageMap::LinesStorageMap(const LineInfo& initial_info) : m_LinesMap{initial_info.getContent()}, m_Name(std::string{"Lines Storage Map"})
{
}

LinesStorageMap::LinesStorageMap(const LineInfo& initial_info, const std::string& name) : m_LinesMap{initial_info.getContent()}, m_Name(name)
{
}

LinesStorageMap::LinesStorageMap(const std::string& name) : m_Name(name)
{
}

void LinesStorageMap::putContent(const LineInfo& target_info)
{
    m_LinesMap.insert(target_info.getContent());
}

const std::unordered_map<std::size_t, std::string>& LinesStorageMap::getContent() const
{
    return m_LinesMap;
}

const std::string& LinesStorageMap::getName() const
{
    return m_Name;
}

File: TextLinesComparing.Library/LinesStorageMap.cs
//-----------------------------------------------------------------------
// <copyright file="LinesStorageMap.cs" company="Demo Projects Workshop">
//     Copyright (c) Demo Projects Workshop. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

#pragma warning disable SA1600 // ElementsMustBeDocumented

namespace TextLinesComparing.Library;

public class LinesStorageMap
{
    private readonly Dictionary<int, string> currentLinesMap;

    public LinesStorageMap()
    {
        this.currentLinesMap = new Dictionary<int, string>();
        this.Name = "Lines Storage Map";
    }

    public LinesStorageMap(LineInfo info, string name = "Lines Storage Map")
    {
        this.currentLinesMap = new Dictionary<int, string>()
        {
            { info.Content.Key, info.Content.Value },
        };

        this.Name = name;
    }

    public string Name { get; set; }

    public Dictionary<int, string> Content
    {
        get { return this.currentLinesMap; }
    }

    public void PutContent(LineInfo target_info)
    {
        this.currentLinesMap.Add(target_info.Content.Key, target_info.Content.Value);
    }
}
